docs/build.md

```diff
@@ -602,6 +602,103 @@ spec:
         name: test-config
 ```
 
+### Defining Triggers
+
+Using the [triggers](https://github.com/shipwright-io/triggers) application you can issue `BuildRun` instances when certain events happen. The idea is to be able to trigger Shipwright builds in a event driven fashion, for that purpose you can watch certain types of events.
+
+The types of events under watch are defined on the `.spec.trigger` attribute, please consider the following example:
+
+```yaml
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+spec:
+  source:
+    url: https://github.com/shipwright-io/sample-go
+    contextDir: docker-build
+    credentials:
+      name: webhook-secret
+  trigger:
+    when: []
+```
+
+Certain types of events will use attributes defined on `.spec.source` to complete the information needed in order to dispatch events.
+
+#### GitHub
+
+The GitHub type is meant to react on events coming from GitHub WebHook, the events are compared against the existing `Build` resources and therefore it can find `Build` based on `.spec.source.url` combined with the attributes on `.spec.trigger.when[].github`.
+
+When `.spec.trigger.when[].github.branches` is empty, the `.spec.source.revision` is employed, the default "branch" name employed is based on the regular source repository configuration.
```

Comment:
Let me start with an example, please. Let's imagine a pull-request is open at the repository X against the `main` branch, it will dispatch a GitHub Event which contains detailed 

Okay, so let me start with a use-case example, please. Let's imagine the user is receiving WebHook events from GitHub, a event comes along with the type `PullRequest` listing the target branch as `main` and as well the repository URL. Based on those, it must then trigger a single `BuildRun` per `Build` object that matches both repository URL and revision, so in other words, there's a direct relationship between one GitHub event and one (or more) `Build` instance.

Given a single GitHub Event must trigger a single `BuildRun` per `Build` instance responsible for that repository, the business rule is basically a fallback approach: probe if the event branch name is listed on `.spec.trigger.when[].github.branches` slice, or, uses the `.spec.source.revision` (by default is assumed "main" when empty).